packages/components/src/drop-zone/index.tsx

```diff
@@ -22,21 +22,51 @@ import {
 	__unstableMotion as motion,
 	__unstableAnimatePresence as AnimatePresence,
 } from '../animation';
+import type { DropType, DropZoneProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
 
-export default function DropZoneComponent( {
+/**
+ * `DropZone` is a component creating a drop zone area taking the full size of its parent element. It supports dropping files, HTML content or any other HTML drop event.
+ *
+ * ```jsx
+ * import { DropZone } from '@wordpress/components';
+ * import { useState } from '@wordpress/element';
+ *
+ * const MyDropZone = () => {
+ *   const [ hasDropped, setHasDropped ] = useState( false );
+ *
+ *   return (
+ *     <div>
+ *       { hasDropped ? 'Dropped!' : 'Drop something here' }
+ *       <DropZone
+ *         onFilesDrop={ () => setHasDropped( true ) }
+ *         onHTMLDrop={ () => setHasDropped( true ) }
+ *         onDrop={ () => setHasDropped( true ) }
+ *       />
+ *     </div>
+ *   );
+ * }
+ * ```
+ */
+export function DropZoneComponent( {
 	className,
 	label,
 	onFilesDrop,
 	onHTMLDrop,
 	onDrop,
-} ) {
-	const [ isDraggingOverDocument, setIsDraggingOverDocument ] = useState();
-	const [ isDraggingOverElement, setIsDraggingOverElement ] = useState();
-	const [ type, setType ] = useState();
+	...restProps
+}: WordPressComponentProps< DropZoneProps, 'div', false > ) {
```

Comment:
Should this component be polymorphic?

Reply:
It could be potentially, but not in the current form because the `div` used in the code is a raw div element and not a polymorphic one (like `View` or `styled.div`).